frinx/workers/http_workers/http_workers.py

```diff
@@ -0,0 +1,83 @@
+from frinx.common.type_aliases import ListStr
+from frinx.common.util import cookie_jar_to_dict
+from frinx.common.util import snake_to_camel_case
+from frinx.common.worker.service import ServiceWorkersImpl
+from frinx.common.worker.task_def import TaskDefinition
+from frinx.common.worker.task_def import TaskExecutionProperties
+from frinx.common.worker.task_def import TaskInput
+from frinx.common.worker.task_def import TaskOutput
+from frinx.common.worker.task_result import TaskResult
+from frinx.common.worker.worker import WorkerImpl
+from frinx.services.http_service import http_service
+from frinx.services.http_service.enums import ContentType
+from frinx.services.http_service.enums import HttpMethod
+from frinx.workers.http_workers.util import BasicAuth
+
+
+class HTTPWorkersService(ServiceWorkersImpl):
+
+    class GenericHTTPWorker(WorkerImpl):
+
+        class WorkerDefinition(TaskDefinition):
+            name: str = 'generic_HTTP_worker'
+            description: str = 'Generic HTTP worker.'
+            labels: ListAny = ['SIMPLE']
+            timeout_seconds: int = 360
+            response_timeout_seconds: int = 360
+
+        class ExecutionProperties(TaskExecutionProperties):
+            transform_string_to_json_valid: bool = True
+            exclude_empty_inputs: bool = True
+        
+        class WorkerInput(TaskInput):
+            uri: AnyHttpUrl
+            method: HttpMethod
+            basic_auth: BasicAuth | None = None
+            content_type: ContentType | None = None
+            body: DictAny | ListAny | str | None = None
+            read_timeout: NonNegativeFloat | None = None
+            connect_timeout: NonNegativeFloat = 360
+            headers: DictStr | None = {}
+            cookies: DictStr | None = {}
+
+            class Config(TaskInput.Config):
+                alias_generator = snake_to_camel_case
+                allow_population_by_field_name = True
+                arbitrary_types_allowed = True
+                use_enum_values = True
+
+        class WorkerOutput(TaskOutput):
+            status_code: int
+            response: DictAny | ListAny | str
+            cookies: DictAny
+            logs: ListStr | str
+
+            class Config(TaskOutput.Config):
+                alias_generator = snake_to_camel_case
+                allow_population_by_field_name = True
+                min_anystr_length = 1
+
+        def execute(self, worker_input: WorkerInput) -> TaskResult[Any]:
+            response = http_service.http_task(worker_input)
+            logs = f'{worker_input.method} {worker_input.uri} {response.status_code} {response.reason}'
+            logs = str(worker_input)
+
+            return TaskResult(
+                logs=logs,
+                status=TaskResultStatus.COMPLETED if response.ok else TaskResultStatus.FAILED,
+                output=self.WorkerOutput(
+                    status_code=response.status_code,
+                    response=response.text,
```

Comment:
Is text suitable in any situation? 
I mean if the response is not a text but JSON or any different type.

Reply:
OK, I complete this with function witch can returns response also in JSON.